enhancements/sig-architecture/229-manifestworkreplicaset-rollback/README.md

```diff
@@ -0,0 +1,434 @@
+    1. Iterate each cluster of `rolloutResult.clusterToRollout`
+        1. `(NEW)` Create `ManifestWork` and add `work.open-cluster-management.io/controller-hash` label with new hash
+        2. Apply `ManifestWork` to each cluster namespace
+    1. Clean up manifestwork resources from each cluster namespace
+    1. `(NEW)` Status update
+        1. Calculate `.summary.updated` by counting the number of `Succeeded` status of rollout clusters (which have the updateRevision)
+        1. if `.summary.updated` == `.summary.desiredTotal`,
+            1. Set `.status.updateRevision` to `.status.currentRevision`
+
+### Status transition
+
+This section describes the detail status transition for each rollout condition.
+
+#### New manifestwork update transition
+
+| PlacementVerified | ManifestworkApplied | Progressing | PlacementRollOut |
+| --- | --- | --- | --- |
+| AsExpected (True) | | NewRevisionCreated (True) | |
+| AsExpected (True)| Progressing (False) | NewRevisionCreated (True) | Progressing (False) | 
+| AsExpected (True)| Progressing (False) | NewRevisionCreated (True) | Progressing (False) | 
+| AsExpected (True)| AsExpected (True) | NewRevisionAvailable (True) | Complete (True) |
+
+#### Rollback (found the existing revision transition)
+
+| PlacementVerified | ManifestworkApplied | Progressing | PlacementRollOut |
+| --- | --- | --- | --- |
+| AsExpected (True) | | FoundNewRevision (True) | |
+| AsExpected (True)| Progressing (False) | FoundNewRevision (True) | Progressing (False) |
+| AsExpected (True)| Progressing (False) | FoundNewRevision (True) | Progressing (False) |
+| AsExpected (True)| AsExpected (True) | NewRevisionAvailable (True) | Progressing (False) |
+| AsExpected (True)| AsExpected (True) | NewRevisionAvailable (True) | Complete (True) |
+
+#### ProgressDeadlineExceeded transition
+
+| PlacementVerified | ManifestworkApplied | Progressing | PlacementRollOut |
+| --- | --- | --- | --- |
+| AsExpected (True) | | NewRevisionCreated (True) | |
+| AsExpected (True)| Progressing (False) | NewRevisionCreated (True) | Progressing (False) | 
+| AsExpected (True)| Progressing (False) | NewRevisionCreated (True) | Progressing (False) | 
+| AsExpected (True)| NotAsExpected (False) | ProgressDeadlineExceeded (False) | RolloutDegraded (False) |
+
+#### Abort transition
+
+| PlacementVerified | ManifestworkApplied | Progressing | PlacementRollOut |
+| --- | --- | --- | --- |
+| AsExpected (True) | | NewRevisionCreated (True) | |
+| AsExpected (True)| Progressing (False) | NewRevisionCreated (True) | Progressing (False) | 
+| AsExpected (True)| Progressing (False) | NewRevisionCreated (True) | Progressing (False) | 
+| AsExpected (True)| NotAsExpected (False) | ProgressDeadlineExceeded (False) | RolloutDegraded (False) |
+| AsExpected (False)| NotAsExpected (False) | RolloutAborted (False) | RolloutDegraded (False) |
+
+### Open Questions
+
+1. Should we have the delay before starting aborting a rollout ?
```

Comment:
sounds reasonable, but we could consider this as an additional abort strategy in beta phase.